src/escrows/VaultHelper.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {IERC4626} from "lib/openzeppelin-contracts/contracts/interfaces/IERC4626.sol";
+import {IERC20} from "lib/openzeppelin-contracts/contracts/interfaces/IERC20.sol";
+import {IMarket} from "src/interfaces/IMarket.sol";
+import {SafeERC20} from "lib/openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+@title Vault Helper
+@notice This contract is a helper contract for an ERC4626 vault contract. 
+It allows users to deposit and withdraw ERC20 tokens from the vault contract and deposit and withdraw shares from the market.
+**/
+contract VaultHelper {
+    using SafeERC20 for IERC20;
+    
+    error InsufficientShares();
+    error AddressZero();
+
+    IMarket public immutable market;
+    IERC20 public immutable token;
+    IERC4626 public immutable vault;
+
+    /** @dev Constructor
+    @param _vault The address of the ERC4626 vault
+    @param _market The address of the market
+    @param _token The address of the ERC20 token
+    **/
+    constructor(address _vault, address _market, address _token) {
+        vault = IERC4626(_vault);
+        market = IMarket(_market);
+        token = IERC20(_token);
+        token.approve(_vault, type(uint).max);
+        vault.approve(_market, type(uint).max);
```

Comment:
Do we need a way to refresh these approvals to prevent a flashmint griefing attack?

Reply:
Fixed in https://github.com/InverseFinance/FiRM/pull/63/commits/5ba1fef715294c0626193dccb933348884351416